hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/sink/ITTestDataStreamWrite.java

```diff
@@ -554,4 +518,20 @@ public void testHoodiePipelineBuilderSinkWithSchemaSet() throws Exception {
     execute(execEnv, false, "Api_Sink_Test");
     TestData.checkWrittenDataCOW(tempFile, EXPECTED);
   }
+
+  private JsonDeserializationFunction getDeserializationFunc(Configuration conf) {
+    // Read from file source
```

Comment:
Nice catch, curious why the CI does not fail for the PR previously, can we move this `#getDeserializationFunc` into `JsonDeserializationFunction` and we can have a static factory method like `JsonDeserializationFunction#getInstance`.

Reply:
> Why CI does not fail

As I understand it happens because azure still runs test with flink 1.14

azure-pipelines.yml:
```
BUILD_PROFILES: '-Dscala-2.11 -Dspark2.4 -Dflink1.14'
```
I was thinking 1.14 version is purposely  left as main for azure but it looks like a bug now :)

> we can have a static factory method

thanks, I'll fix it

Reply:
Yeah, we actually should use the latest flink version for Azure tests.

Reply:
- Created  factory method
- Updated azure for flink 1.16